### ISSUE TITLE
vscode.env.openExternal handles plus sign incorrectly

### ISSUE THREAD

**Issue by GYK90** [microsoft/vscode#124704]
Issue Type: Bug

## problem summary
I want to open a url with parameters including plus sign. But after the web is opened, plus sign is missing.

## Reproduce steps:
1. in your browser, paste "https://www.google.com/search?q=1+%2B+1+%3C+3" into the address bar, press 'enter'. This should open Google and it searches "1 + 1 < 3" in the search bar.
2. in a webview of extension, add 
    ```
    vscode.env.openExternal(vscode.Uri.parse("https://www.google.com/search?q=1+%2B+1+%3C+3")) 
   ```
and run this code. 
### Expected:
When the code is executed, google is opened and "1 + 1 < 3" in the search bar.
### Actual result:
Google is open but "1   1 < 3" in the search bar.

VS Code version: 1.50.2021051101
OS version: macOS Big Sur version 11.3.1 (20E241)

**Comment by GYK90** [comment]
**Our work is blocked by this issue and many users complain about it.**

What I found is:  the vscode.Uri.parse(url) will convert all space to "+".  And vscode.env.openExternal() will interpret all "+" as space. So the real "+" will also be interpreted as a space.

**Comment by mjbvz** [comment]
@jrieken I investigated this but don't know how to fix it without potentially causing other regressions.

At first I thought extensions may be able to workaround this issue using `Uri.from`:

```ts
        const x = vscode.Uri.parse("https://www.google.com/search").with({ query: 'q=1+%2B+1+%3C+3' });
```

But this results in us opening `https://www.google.com/search?q=1+%252B+1+%253C+3` (which is the query `1 %2B 1 %3C 3` on google)
 
Here's where the incorrect double encoding takes places:

https://github.com/microsoft/vscode/blob/a6f7aa5e4c5832c3d788ab9a8146ac69a6e5a77f/src/vs/editor/browser/services/openerService.ts#L228


This may fall under #85930. We could also consider allowing extensions call `openExternal` with a string that we don't try to modify

**Comment by jrieken** [comment]
/duplicate of https://github.com/microsoft/vscode/issues/85930
